File: frontend/auction-app/src/Services/subcategory-service.js
import BaseService from './BaseService'
 
export default class SubCategoryService extends BaseService {

     getRandomSubCategories(n) {
        return this.baseApi({
            method: 'GET',
            url: `/subcategories/random/${n}`
        });
    }

    getAllSubCategories() {
        return this.baseApi({
            method: 'GET',
            url: '/subcategories/all'
        })
    }

    getSubcategoriesByCategoryId(id) {
        return this.baseApi({
            method: 'GET',
            url: `/subcategories/${id}`
        });
    }
}

File: backend/src/main/java/com/atlantbh/auction/service/SubCategoryService.java
package com.atlantbh.auction.service;

import com.atlantbh.auction.exceptions.RepositoryException;
import com.atlantbh.auction.exceptions.ServiceException;
import com.atlantbh.auction.model.SubCategory;
import com.atlantbh.auction.model.filter.NullFilterBuilder;
import com.atlantbh.auction.repository.SubCategoryRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class SubCategoryService extends BaseService<SubCategory, Long, NullFilterBuilder, SubCategoryRepository> {

    public List<SubCategory> getRandomSubcategories(int n) throws ServiceException {
        try {
            return repository.getRandomSubcategories(n);
        } catch (RepositoryException e) {
            throw new ServiceException("There was an issue with returning the random subcategories", e);
        }
    }

    public List<SubCategory> getAllSubcategories() throws ServiceException {
        try {
            return repository.getSubcategories();
        } catch (RepositoryException e) {
            throw new ServiceException("There was an issue with returning the subcategories", e);
        }
    }
}
